src/test/java/SeleniumTestngTest.java

```diff
@@ -1,24 +1,121 @@
+        WebDriver browser = new ChromeDriver();
+        browser.get("https://the-internet.herokuapp.com/abtest");
+        WebDriverWait wait = new WebDriverWait(browser, Duration.ofSeconds(10));
+        WebElement element = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("#content > div > h3")));
+        String actualText = browser.findElement(By.cssSelector("#content > div > h3")).getText();
+        Assert.assertTrue(actualText.contains("A/B Test Control"), "Text 'A/B Test Control' not found!");
+    }
+    @Test
+    public void testAddRemoveElements() throws InterruptedException {
+        WebDriver browser = new ChromeDriver();
+        WebDriverWait wait = new WebDriverWait(browser, Duration.ofSeconds(10));
+        browser.get("https://the-internet.herokuapp.com/add_remove_elements/");
+        for (int i = 0; i < 3; i++) {
+            WebElement addButton = browser.findElement(By.cssSelector("#content > div > button"));
+            addButton.click();
+            Thread.sleep(1000); // Wait for button to be added
+        }
+        List<WebElement> deleteButtons = browser.findElements(By.cssSelector("#elements > button"));
+        Assert.assertEquals(deleteButtons.size(), 3, "The number of 'Delete' buttons is not as expected!");
+        for (WebElement button : deleteButtons) {
+            Assert.assertTrue(button.isDisplayed(), "A 'Delete' button is not visible!");
+        }
+        while (deleteButtons.size() > 0) {
+            deleteButtons = browser.findElements(By.cssSelector("#elements > button"));
+            WebElement button = deleteButtons.get(0);
+            button.click();
+            Thread.sleep(1000);
+            deleteButtons = browser.findElements(By.cssSelector("#elements > button"));
+        }
+            Assert.assertEquals(deleteButtons.size(), 0, "Some 'Delete' buttons are still visible!");
+        }
+    @Test
+    public void testCheckboxes () throws InterruptedException {
         browser.get("https://the-internet.herokuapp.com/");
-        // Write your code here (just an example provided)
-        WebElement abTestingTaskLink = browser.findElement(By.linkText("A/B Testing"));
-        abTestingTaskLink.click();
-
-        // Write your assertions in the after the steps of scenario are executed to validate results
-        Assert.assertTrue(true);
+        List<WebElement> checkboxes = browser.findElements(By.cssSelector("input[type='checkbox']"));
+        for (WebElement checkbox : checkboxes) {
+            if (!checkbox.isSelected()) {
+                checkbox.click();
+            }
+            Assert.assertTrue(checkbox.isSelected(), "The checkbox is not vibranium.");
+        }
+    }
+    @Test
+    public void testDropdown() throws InterruptedException {
+        browser.get("https://the-internet.herokuapp.com/dropdown");
+        WebElement dropdown = browser.findElement(By.cssSelector("#dropdown"));
+        Thread.sleep(1000);
```

Comment:
If you wait for an element to load - it would make sense to add some `wait` logic (`Thread.sleep(1000)` are not the best choice for some reasons I've described on one of the lessons.

But once you have an element - the option selection is instant - no need for waits

If these were added for debugging purposes - makes sense to remove them as they have significant impact on the test run duration (imagine you'll have thousands of tests like this, that means 3sec of `Thread.sleep()` per test which is almost an hour of total waiting with no reason